### PR TITLE
fix(protocol): use of long instead of time objects

### LINKED ISSUE #2624: JobRecord and JobBatchRecord should return long
**BREAKING CHANGE**

Consolidate everything to long and document the unit, i.e. duration in ms or timestamp since unix epoch

See https://github.com/zeebe-io/zeebe/issues/2616

### PR DESCRIPTION
**Pull request by ChrisKujawa**
closes #2624

### CODE REVIEW

File: protocol-impl/src/main/java/io/zeebe/protocol/impl/record/value/job/JobBatchRecord.java [hunk 1]
```diff
@@ -83,8 +82,9 @@ public JobBatchRecord setWorker(DirectBuffer worker, int offset, int length) {
     return variablesProp;
   }
 
-  @JsonProperty("timeout")
-  public long getTimeoutLong() {
+  @Override
+  @JsonProperty
```
Comment (saig0) [inline protocol-impl/src/main/java/io/zeebe/protocol/impl/record/value/job/JobBatchRecord.java]: I think you can remove `@JsonProperty` annotation now. Just, one method left

File: protocol-impl/src/main/java/io/zeebe/protocol/impl/record/value/job/JobRecord.java [hunk 2]
```diff
@@ -76,8 +75,9 @@ public DirectBuffer getCustomHeadersBuffer() {
     return customHeadersProp.getValue();
   }
 
-  @JsonProperty("deadline")
-  public long getDeadlineLong() {
+  @Override
+  @JsonProperty
```
Comment (saig0) [inline protocol-impl/src/main/java/io/zeebe/protocol/impl/record/value/job/JobRecord.java]: I think you can remove `@JsonProperty` annotation now. Just, one method left

File: protocol/src/main/java/io/zeebe/protocol/record/value/JobRecordValue.java [hunk 3]
```diff
@@ -47,11 +46,11 @@
...
   /**
-   * @return the time until when the job is exclusively assigned to this worker. If the deadline is
-   *     exceeded, it can happen that the job is handed to another worker and the work is performed
-   *     twice.
+   * @return the unix timestamp until when the job is exclusively assigned to this worker (time unit
+   *     is milliseconds since unix epoch). If the deadline is exceeded, it can happen that the job
+   *     is handed to another worker and the work is performed twice.
    */
-  Instant getDeadline();
+  long getDeadline();
```
Comment (saig0) [inline protocol/src/main/java/io/zeebe/protocol/record/value/JobRecordValue.java]: You should say that it can be `-1` if the job is not activated.

**Review by saig0 — CHANGES_REQUESTED** [review]
Found some small things ;)

**Comment by Zelldon** [comment]
You're right. I fixed this

**Review by saig0 — APPROVED** [review]

**Pull request merged**